make/src/special_target.rs

```diff
@@ -211,4 +250,80 @@ impl Processor<'_> {
 
         Ok(())
     }
+
+    fn process_suffixes(self) -> Result<(), Error> {
+        let suffixes_key = ".SUFFIXES";
+        let suffixes_set = self
+            .rule
+            .prerequisites()
+            .map(|suffix| suffix.as_ref().to_string())
+            .collect::<BTreeSet<String>>();
+
+        if suffixes_set.is_empty() {
+            self.make
+                .config
+                .rules
+                .insert(suffixes_key.to_string(), BTreeSet::new());
+        } else {
+            self.make
+                .config
+                .rules
+                .insert(suffixes_key.to_string(), suffixes_set);
+        }
```

Comment:
This check seems unnecessary, simplify by 
```suggestion
        self.make
            .config
            .rules
            .insert(suffixes_key.to_string(), suffixes_set);
```